src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const separatorString = String(separator);
+    const lastElement = this[this.length - 1];
+    let result = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        result += separatorString;
+      } else {
+        result += this[i] + separatorString;
+      }
+    }
+
+    if (lastElement !== undefined) {
+      result += lastElement;
+    }
+
+    return result;
```

Comment:
Maybe, it's better not to use 'result' as a name for variable
<sub>Just my opinion</sub>